test/unit/support/shared/virtualbox_context.rb

```diff
@@ -28,6 +28,11 @@ def subprocess_result(options={})
     allow(subprocess).to receive(:execute).
       with("VBoxManage", "showvminfo", kind_of(String), kind_of(Hash)).
       and_return(subprocess_result(exit_code: 0))
+
+    # apparently this is also used in docker tests
+    allow(subprocess).to receive(:execute).
+      with("docker", "version", an_instance_of(Hash)).
+      and_return(subprocess_result(exit_code: 0))
```

Comment:
This is still not required. We should remove the `include_context "virtualbox"` lines in:

- test/unit/plugins/providers/docker/action/connect_networks_test.rb
- test/unit/plugins/providers/docker/action/destroy_network_test.rb
- test/unit/plugins/providers/docker/action/login_test.rb
- test/unit/plugins/providers/docker/action/prepare_networks_test.rb

As I believe I included it by mistake. If you look at the stacktrace from the failing test, it's coming from that VirtualBox shared context test class. Docker should not be here, this is a virtualbox class.

Reply:
I did try without this first. Resulting failures: CircleCI [ruby23](https://circleci.com/gh/hashicorp/vagrant/475), [ruby24](https://circleci.com/gh/hashicorp/vagrant/478), [ruby25](https://circleci.com/gh/hashicorp/vagrant/477), [ruby26](https://circleci.com/gh/hashicorp/vagrant/476); [Travis](https://travis-ci.org/hashicorp/vagrant/builds/583372763).

Fundamentally, adding a working usability check to the Docker provider does mean that all the other Docker provider tests will run it, so it needs to be mocked.

Reply:
Perhaps the “virtualbox” shared context should just be renamed to “subprocess” or something?